amd_openvx_extensions/amd_rpp/source/tensor/ColorTwist.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct ColorTwistLocalData {
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    Rpp32f *pAlpha;
+    Rpp32f *pBeta;
+    Rpp32f *pHue;
+    Rpp32f *pSat;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptROI *pSrcRoi;
+    RpptRoiType roiType;
+    vxTensorLayout inputLayout;
+    vxTensorLayout outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+static vx_status VX_CALLBACK refreshColorTwist(vx_node node, const vx_reference *parameters, vx_uint32 num, ColorTwistLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[3], 0, data->inputTensorDims[0], sizeof(Rpp32f), data->pAlpha, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
```

Comment:
I have mentioned this before. We need to pass this as a single tensor ptr with 4 float values for (alpha, beta, hue and sat). This was we can pass the same ptr (host/device) to RPP without copying

Reply:
Hi @rrawther 
We discussed about the changes required for this in call and we decided not to do this.
Below were the pointers discussed.
Currently we have all of these variables as vx_array in rocAL as they are generated using the Parameter class. Internally it accepts batch size and generates random or fixed values and fills the vx_array.
In order to convert to vx_tensor we need to create a tensor fill with all the array values and again at the end when we need to pass to RPP we must split it into the 4 arrays.
Also in cases where we have multiple inputs with different data types we might not be able to do it.